tidy3d/web/api/webapi.py

```diff
@@ -88,15 +89,35 @@ def run(
         verbose=verbose,
         progress_callback=progress_callback_upload,
     )
-    start(
-        task_id,
-        solver_version=solver_version,
-        worker_group=worker_group,
-    )
-    monitor(task_id, verbose=verbose)
-    return load(
-        task_id=task_id, path=path, verbose=verbose, progress_callback=progress_callback_download
-    )
+    try:
+        start(
+            task_id,
+            solver_version=solver_version,
+            worker_group=worker_group,
+        )
+        monitor(task_id, verbose=verbose)
+        return load(
+            task_id=task_id,
+            path=path,
+            verbose=verbose,
+            progress_callback=progress_callback_download,
+        )
+    except TaskRepeatedError as e:
```

Comment:
when does this get raised? in `start`? if so, maybe we should put this `except` handling there?  otherwise it will only have this feature if user uses `web.run()`,  not if they do `web.start()` nor if they use `Job.run()`, `Batch.run()`. I think it would be better if we could catch it at all times?

Maybe the issue is that we need a new task_id to continue the loading? if that's the case, maybe a cleaner approach is to allow `web.start` to return an optional `task_id` object and then slightly refactor the webapi internals to use this returned task_id in the subsequent code?

Basically in it's current stage, I think it needs to be refactored to future-proof and make this generally work every time `web.start()` is called.